### PR TITLE
feat(renderState): add connectNumericMenu

### PR DESCRIPTION
**Pull request by Haroenv**
DX-204

### CODE REVIEW

File: src/connectors/numeric-menu/__tests__/connectNumericMenu-test.ts [hunk 1]
```diff
@@ -1017,4 +1017,218 @@ See documentation: https://www.algolia.com/doc/api-reference/widgets/numeric-men
...
+          helper,
+          state: helper.state,
+          results,
+        })
+      );
+
+      expect(renderState2.numericMenu).toEqual({
+        numerics: {
+          createURL: expect.any(Function),
+          refine: expect.any(Function),
```
Comment (yannickcr) [inline src/connectors/numeric-menu/__tests__/connectNumericMenu-test.ts:1089]: Maybe use:

```suggestion
          refine: renderState1.numericMenu.numerics.refine,
```

Just to be sure we get the same function as in the first render ?

File: src/connectors/numeric-menu/__tests__/connectNumericMenu-test.ts [hunk 2]
```diff
@@ -1017,4 +1017,218 @@ See documentation: https://www.algolia.com/doc/api-reference/widgets/numeric-men
...
+            label: '10 - 20',
+            value: '%7B%22start%22:10,%22end%22:20%7D',
+          },
+          {
+            isRefined: false,
+            label: 'more than 20',
+            value: '%7B%22start%22:20%7D',
+          },
+        ],
+        refine: expect.any(Function),
```
Comment (yannickcr) [inline src/connectors/numeric-menu/__tests__/connectNumericMenu-test.ts:1212]:
```suggestion
        refine: renderState1.refine,
```

**Review by yannickcr — APPROVED** [review]

File: src/connectors/numeric-menu/__tests__/connectNumericMenu-test.ts [hunk 3]
```diff
@@ -812,7 +812,7 @@ See documentation: https://www.algolia.com/doc/api-reference/widgets/numeric-men
       });
     });
 
-    test('returns the `uiState` with a refinement (range)', () => {
+    test('returns the `uiState` with a refinement (numeric)', () => {
```
Comment (Haroenv) [inline src/connectors/numeric-menu/__tests__/connectNumericMenu-test.ts:815]:
```suggestion
    test('returns the `uiState` with a refinement (range)', () => {
```

File: src/connectors/numeric-menu/__tests__/connectNumericMenu-test.ts [hunk 4]
```diff
@@ -970,7 +970,7 @@ See documentation: https://www.algolia.com/doc/api-reference/widgets/numeric-men
       );
     });
 
-    test('returns the `SearchParameters` with the value from `uiState` (range)', () => {
+    test('returns the `SearchParameters` with the value from `uiState` (numeric)', () => {
```
Comment (Haroenv) [inline src/connectors/numeric-menu/__tests__/connectNumericMenu-test.ts:973]:
```suggestion
    test('returns the `SearchParameters` with the value from `uiState` (range)', () => {
```

**Review by shortcuts — APPROVED** [review]

**Pull request merged**